### PR TITLE
smpeg2: 412 -> unstable-2017-10-18

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

* rework of https://github.com/NixOS/nixpkgs/pull/70421
* updated by 1 SVN revision
* use new format for version
* drop patch not needed anymore
* rename gcc6.patch to hufftable-uint_max.patch to make it more obvious what it does

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105216` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>frogatto</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>devilutionx</li>
    <li>hedgewars</li>
    <li>worldofgoo</li>
  </ul>
</details>
<details>
  <summary>76 packages built:</summary>
  <ul>
    <li>SDL2_mixer</li>
    <li>Sylk</li>
    <li>_1oom</li>
    <li>abbaye-des-morts</li>
    <li>appimage-run</li>
    <li>augustus</li>
    <li>cataclysm-dda</li>
    <li>cataclysm-dda-git</li>
    <li>chocolateDoom</li>
    <li>commandergenius</li>
    <li>crawlTiles</li>
    <li>crispyDoom</li>
    <li>deltachat-electron</li>
    <li>devdocs-desktop</li>
    <li>easyrpg-player</li>
    <li>eduke32</li>
    <li>electronplayer</li>
    <li>flare</li>
    <li>gemrb</li>
    <li>guile-sdl2</li>
    <li>haskellPackages.Raincat</li>
    <li>haskellPackages.nyx-game</li>
    <li>haskellPackages.piyo</li>
    <li>haskellPackages.sdl2-mixer</li>
    <li>instead</li>
    <li>instead-launcher</li>
    <li>irccloud</li>
    <li>ivan</li>
    <li>joplin-desktop</li>
    <li>julius</li>
    <li>keeweb</li>
    <li>ledger-live-desktop</li>
    <li>liberal-crime-squad</li>
    <li>lightspark</li>
    <li>lunar-client</li>
    <li>marktext</li>
    <li>minetime</li>
    <li>molotov</li>
    <li>mycrypto</li>
    <li>notable</li>
    <li>np2kai</li>
    <li>nuclear</li>
    <li>nxengine-evo</li>
    <li>opendune</li>
    <li>p3x-onenote</li>
    <li>pingus</li>
    <li>plexamp</li>
    <li>python27Packages.pygame_sdl2</li>
    <li>python27Packages.pysdl2</li>
    <li>python27Packages.pytmx</li>
    <li>python37Packages.pygame_sdl2</li>
    <li>python37Packages.pysdl2</li>
    <li>python37Packages.pytmx</li>
    <li>python38Packages.pygame_sdl2</li>
    <li>python38Packages.pysdl2</li>
    <li>python38Packages.pytmx</li>
    <li>renpy</li>
    <li>retrofe</li>
    <li>rocksndiamonds</li>
    <li>runwayml</li>
    <li>sfrotz</li>
    <li>smpeg2</li>
    <li>ssb-patchwork</li>
    <li>standardnotes</li>
    <li>station</li>
    <li>steam-run-native</li>
    <li>taisei</li>
    <li>timeular</li>
    <li>trackballs</li>
    <li>tusk</li>
    <li>ultrastardx</li>
    <li>unityhub</li>
    <li>wesnoth (wesnoth-dev)</li>
    <li>wootility</li>
    <li>zettlr</li>
    <li>zulip</li>
  </ul>
</details>

File: pkgs/development/libraries/smpeg2/default.nix [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
   src = fetchsvn {
     url = "svn://svn.icculus.org/smpeg/trunk";
-    rev = version;
-    sha256 = "1irf2d8f150j8cx8lbb0pz1rijap536crsz0mw871xrh6wd2fd96";
+    rev = "413";
+    sha256 = "193amdwgxkb1zp7pgr72fvrdhcg3ly72qpixfxxm85rzz8g2kr77";
   };
 
   patches = [
-    ./gcc6.patch
```
Comment (Mic92) [inline pkgs/development/libraries/smpeg2/default.nix:14]: please also delete this one.

Comment (prusnak) [reply pkgs/development/libraries/smpeg2/default.nix:14]: I am not sure what you mean by this as you are commenting on a deleted line and the file `gcc6.patch` has been renamed to `hufftable-uint_max.patch`. Are you suggesting to drop the patch completely?

Comment (prusnak) [reply pkgs/development/libraries/smpeg2/default.nix:14]: Just to be sure, I tried to remove the `hufftable-uint_max.patch` and the build fails with

```
audio/hufftable.cpp:587:1: error: narrowing conversion of '-1' from 'int' to 'unsigned int' [-Wnarrowing]
```

Comment (Mic92) [reply pkgs/development/libraries/smpeg2/default.nix:14]: Right. I did not spot the rename.

**Pull request merged**